packages/grafana-ui/src/components/Segment/Segment.story.tsx

```diff
@@ -37,7 +37,38 @@ SegmentStories.add('Array Options', () => {
             />
             <Segment
               Component={AddButton}
-              onChange={(value: SelectableValue<string>) => action('New value added')(value)}
+              onChange={value => action('New value added')(value as SelectableValue<string>)}
+              options={options}
+            />
+          </div>
+        </>
+      )}
+    </UseState>
+  );
+});
+
+SegmentStories.add('Array Options With Primitive value', () => {
+  const options = ['Option1', 'Option2', 'OptionWithLooongLabel', 'Option4'].map(toOption);
+  options[0].label = 'Option1 Label';
+  return (
+    <UseState initialState={options[0].value}>
```

Comment:
I think it's time to start using useState hook for this :)

Reply:
yes!